src/v2/recommender.py

```diff
@@ -4,474 +4,202 @@
 sourcing.
 """
 
```

Comment:
This file has completed changed, recommended to look at the file directly rather than the diff.